ols/app/endpoints/ols.py

```diff
@@ -234,7 +233,22 @@ def generate_bare_response(conversation_id: str, llm_request: LLMRequest) -> str
 
     prompt = PromptTemplate.from_template("{query}")
     llm_chain = LLMChain(llm=bare_llm, prompt=prompt, verbose=True)
-    response = llm_chain(inputs={"query": llm_request.query})
+    token_counter = GenericTokenCounter(bare_llm)
```

Comment:
This code is more or less repeated in 3 places. It would change the implementation, but what about having it as a context manager? Something like:
```python
with metrics.token_counter(bare_llm):
   llm_chain = LLMChain(llm=bare_llm, prompt=prompt, verbose=True)   
   response = llm_chain(inputs={"query": llm_request.query})
```

Reply:
I will change GenericTokenCounter that it updates the metrics by the end of the callback function.